### PR TITLE
{tools}[fosscuda/2018b] gpustat v0.5.0 w/ Python 2.7.15

### PR DESCRIPTION
**Pull request by adiemu5**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by adiemu5** [comment]
Test report by @adiemu5
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
gpu01.wave.scu.edu - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/46cf6a8283606e8531ca265050853d69 for a full test report.

**Comment by adiemu5** [comment]
Test report by @adiemu5
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
gpu01.wave.scu.edu - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/ee316ba49b6a66bb817b034539ec141d for a full test report.

**Comment by adiemu5** [comment]
I'm rather new to EasyBuild and at a loss for how to proceed here.

I *think* the initial PR failed due to pulling in undeclared dependencies.  Rather than creating separate easyconfigs for all of them, I switched to a bundle, where they're all declared.  This works locally, but the pipeline doesn't like them for reasons I don't really understand.

I don't set 'use_pip' anywhere, so I don't know why that's mentioned in both CI run reports.  I added 'download_dep_fail' explicitly in the hope that it would fail locally if anything was omitted, rather than having to wait for another CI run to find it.  But if I read the second CI log correctly, it failed *because* I had 'download_dep_failed' set on a PythonBundle?

Happy to fix this, but I really don't understand what needs to be done, and I don't want to clutter up the PR just trying things at random.

**Comment by adiemu5** [comment]
Test report by @adiemu5
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
gpu01.wave.scu.edu - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/e6d6dee8c273615b0853bdce7c5c9125 for a full test report.

**Comment by adiemu5** [comment]
Test report by @adiemu5
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
gpu01.wave.scu.edu - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/788bac9039cb07f975fba27fdd792535 for a full test report.

**Comment by jose-d** [comment]
Test report by @jose-d
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
n28 - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz, Python 2.7.5
See https://gist.github.com/0cf3a44246ec377091fd90642235d324 for a full test report.

**Comment by jose-d** [comment]
```
jose@n28:~$ module load tools/gpustat/0.5.0-fosscuda-2018b-Python-2.7.15
jose@n28:~$ gpustat 
n28  Mon Apr  8 12:33:34 2019
[0] Tesla P100-SXM2-16GB | 37'C,   0 % |     0 / 16280 MB |
[1] Tesla P100-SXM2-16GB | 32'C,   0 % |     0 / 16280 MB |
[2] Tesla P100-SXM2-16GB | 30'C,   0 % |     0 / 16280 MB |
[3] Tesla P100-SXM2-16GB | 36'C,   5 % |     0 / 16280 MB |
jose@n28:~$
```
looks fine.

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
grc-cluster1 - Linux centos 6.10, Intel(R) Xeon(R) CPU E5-2640 0 @ 2.50GHz, Python 2.7.14
See https://gist.github.com/b215980afdf9a8ed9bc01966694c8831 for a full test report.

File: easybuild/easyconfigs/g/gpustat/gpustat-0.5.0-fosscuda-2018b-Python-2.7.15.eb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'https://github.com/wookayin/gpustat'
+description = 'dstat-like utilization monitor for NVIDIA GPUs'
+
+toolchain = {'name': 'fosscuda', 'version': '2018b'}
+
+dependencies = [
+    ('Python', '2.7.15'),
+    ('CUDA', '9.2.88'),
```
Comment (migueldiascosta) [inline easybuild/easyconfigs/g/gpustat/gpustat-0.5.0-fosscuda-2018b-Python-2.7.15.eb]: this is not necessary, it's already included in the `fosscuda` toolchain

Comment (adiemu5) [reply easybuild/easyconfigs/g/gpustat/gpustat-0.5.0-fosscuda-2018b-Python-2.7.15.eb]: Fixed

**Comment by boegel** [comment]
@adiemu5 The tests are checking a couple of things, i.e.:

i) that `use_pip` is enabled (because we're trying to enabling `use_pip` by default
ii) that `download_dep_fail` is enabled, to ensure that the easyconfig is not downloading additional Python packages during installation (side note: `download_dep_fail` is enabled by default when using `PythonBundle`, but not (yet) when using `PythonPackage`)

The `AssertError` messages can be a bit confusing, since they report what is expected, not what is found (hence the assertion fails).

Thanks a lot for your efforts on making this pass our picky tests!

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2102.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/564968554c028bf8aa6fd1a3aee04316 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @adiemu5!

**Pull request merged**